kratos/geometries/point_on_geometry.h

```diff
@@ -0,0 +1,378 @@
+
+    typedef GeometryData::IntegrationMethod IntegrationMethod;
+
+    typedef typename BaseType::IndexType IndexType;
+    typedef typename BaseType::SizeType SizeType;
+
+    typedef typename BaseType::PointsArrayType PointsArrayType;
+    typedef typename BaseType::CoordinatesArrayType CoordinatesArrayType;
+    typedef typename BaseType::GeometriesArrayType GeometriesArrayType;
+    typedef typename BaseType::IntegrationPointsArrayType IntegrationPointsArrayType;
+
+    static constexpr IndexType BACKGROUND_GEOMETRY_INDEX = -1;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor for untrimmed patch
+    PointOnGeometry(
+        const CoordinatesArrayType LocalCoordinates,
+        typename GeometryType::Pointer pBackgroundGeometry)
+        : BaseType(PointsArrayType(), &msGeometryData)
+        , mLocalCoordinates(LocalCoordinates)
+        , mpBackgroundGeometry(pBackgroundGeometry)
+    {
+        KRATOS_ERROR_IF(pBackgroundGeometry->WorkingSpaceDimension() != this->WorkingSpaceDimension())
+            << "Working space dimension of background geometry (" << pBackgroundGeometry->WorkingSpaceDimension()
+            << ") does not coincide with WorkingSpaceDimension of this geometry (" << this->WorkingSpaceDimension() << ")."
+            << std::endl;
+        KRATOS_ERROR_IF(pBackgroundGeometry->LocalSpaceDimension() != this->LocalSpaceDimension())
+            << "Local space dimension of background geometry (" << pBackgroundGeometry->LocalSpaceDimension()
+            << ") does not coincide with LocalSpaceDimension of this geometry (" << this->LocalSpaceDimension() << ")."
+            << std::endl;
+    }
+
+    /// Copy constructor.
+    PointOnGeometry(PointOnGeometry const& rOther )
+        : BaseType( rOther )
+        , mLocalCoordinates(rOther.mLocalCoordinates)
+        , mpBackgroundGeometry(rOther.mpBackgroundGeometry)
+    {
+    }
+
+    /// Destructor
+    ~PointOnGeometry() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// Assignment operator.
+    PointOnGeometry& operator=( const PointOnGeometry& rOther )
+    {
+        BaseType::operator=( rOther );
+        mLocalCoordinates = rOther.mLocalCoordinates;
+        mpBackgroundGeometry = rOther.mpBackgroundGeometry;
+        return *this;
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    typename BaseType::Pointer Create( PointsArrayType const& ThisPoints ) const override
+    {
+        return typename BaseType::Pointer( new PointOnGeometry( ThisPoints ) );
+    }
+
+    ///@}
+    ///@name Access to Geometry Parts
+    ///@{
+
+    /**
+    * @brief This function returns the pointer of the background geometry
+    *        of this point. Accesseable via BACKGROUND_GEOMETRY_INDEX.
+    * @param Index: only possible with BACKGROUND_GEOMETRY_INDEX.
+    * @return pointer of geometry, corresponding to the index.
+    */
+    GeometryPointer pGetGeometryPart(IndexType Index) override
```

Comment:
@tteschemacher do we have the same behaviour with curve on surface and others? I mean giving the background geometry as geometry part?

Reply:
yes, we do always return the geometry containing the degrees of freedom with the index -1. All indices above are related to assigned indices.

Reply:
This feature is actually used from time to time. Especially in the BrepSurface.